zebra-state/src/service/finalized_state/zebra_db/shielded.rs

```diff
@@ -355,6 +396,14 @@ impl DiskWriteBatch {
             self.zs_insert(&orchard_tree_cf, height, trees.orchard);
         }
 
+        if let Some(subtree) = trees.sapling_subtree {
+            self.zs_insert(&sapling_subtree_cf, subtree.index, subtree);
+        }
+
+        if let Some(subtree) = trees.orchard_subtree {
+            self.zs_insert(&orchard_subtree_cf, subtree.index, subtree);
+        }
```

Comment:
To test that this code works after the Sapling or NU5 network upgrades, run a state validity check every time Zebra starts up. Here's an example:
https://github.com/ZcashFoundation/zebra/pull/7365/files#diff-1c066eab26934db6288dddda08c0c7030c8399c4b17a70892b2e988aa6404e90R453-R454

Reply:
I'd prefer to only increment the database version once for adding subtrees and to only write this test for that version.

I reverted the database format minor version and commented out subtree insertions in https://github.com/ZcashFoundation/zebra/pull/7334/commits/0d974ac578730cb7bb39cb3203d2c5c903b8e214.

Reply:
Ok, that also works. I'll try merging this now, but the next PR will have to wait until PRs https://github.com/ZcashFoundation/zebra/pull/7385 and https://github.com/ZcashFoundation/zebra/pull/7379 merge.